frappe/model/document.py

```diff
@@ -1093,13 +1093,21 @@ def run_before_save_methods(self):
 	def load_doc_before_save(self, *, raise_exception: bool = False):
 		"""load existing document from db before saving"""
 
+		for_update = True
+		if (
+			getattr(self.meta, "issingle", True)
```

Comment:
Umm, 2 things

Keep framework flags as generic as possible. Why should this only apply to singles? There are other single doctypes where it should definitely lock the document.


Here's suggestion:

1. Assume for_update = true always
2. If flag is not None then use that value instead. I.e. developer has explicitly set it.

Add kwarg in run_server_method to not lock the document for special cases like payment reco tool.